packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -656,7 +681,6 @@ export class BaseResolversVisitor<
   protected _globalDeclarations = new Set<string>();
   protected _federation: ApolloFederation;
   protected _hasScalars = false;
-  protected _hasFederation = false;
```

Comment:
Previously, whenever there's a type with `__resolveReference`, this will be set to `true`.

Instead of a `boolean`, we now want to track all the types with generated `__resolveReference`  types as output `meta`, so downstream consumers (e.g. Server Preset) can easily use it. 
The `ApolloFederation` class now tracks this metadata. 

I could just put it here as a property of the plugin e.g. `_federationMeta` but it seems like `ApolloFederation` is supposed to handle federation concerns. Happy to discuss if needed.